build.gradle

```diff
@@ -62,6 +63,29 @@ spotless {
 
 jar { dependsOn checkLicenses }
 
+class FileStateChecker extends DefaultTask {
+    Set<File> files
+    String knownHash
+
+    @TaskAction
+    def CheckState() {
+        def allContents = this.files.sort()*.readLines()*.join('\n').join('')
```

Comment:
It would probably be better to read the files as binary streams and feed them into the `MessageDigest` incrementally rather than reading every file as a string then converting back to bytes all in memory.